common-app-lib/src/cmr/common_app/app_events.clj

```diff
@@ -0,0 +1,74 @@
+(ns cmr.common-app.app-events
+  "Common events which applications can use when starting and stopping."
+  (:require
+   [clojure.java.io :as io]
+   [cmr.common-app.config :as config]
+   [cmr.common.log :refer [debugf infof warnf errorf]]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(import 'com.sun.management.HotSpotDiagnosticMXBean)
+(import 'java.lang.management.ManagementFactory)
+
+;;
+;; The internet suggests the following as how one would write the dump file in java:
+;;
+;;public static void dumpHeap (String filePath, boolean live) throws IOException {
+;;  MBeanServer server = ManagementFactory.getPlatformMBeanServer ();
+;;  HotSpotDiagnosticMXBean mxBean = ManagementFactory.newPlatformMXBeanProxy (server,
+;;      "com.sun.management:type=HotSpotDiagnostic", HotSpotDiagnosticMXBean.class);
+;;   mxBean.dumpHeap (filePath, live);
+;;}
+
+(defn- dump-heap
+  "Write a diagnostics file to the file system for analysis in Visual VM"
+  [^String app-name ^String file-path ^java.lang.Boolean live]
```

Comment:
Am I correct that the file-path is configuration driven only and so the name of the resulting dump file is not built from the app-name, or am I overlooking that here? It looks like the app-name (set as instance-name below I think?) is only used for the message events?
This might be a question for the follow-up ticket (where we write the dump file to some external and accessible location) but thinking that we probably want the dump file name to be unique and identify the source task; you already create the unique (ish) instance-name.

Reply:
app-name is used for the logs to report which app (search vs ingest) is doing this. I have to have this as a parameter because I put this code in common-app-lib instead of having a copy in each app.

file-path is passed in and decided by the caller.

Reply:
Yeah, I think my point was that we are writing the same file name, which is defined by a configuration item rather than creating a coredump file name, dynamically, which serves to identify the app source. So every dump file, created from each of the tasks on every shutdown will have the same name.

Reply:
yes, that is a problem locally, I can change that

Reply:
Hm, well, ok, that was kinda my question. Are you using the app specific name as part of the target file? Bc it doesn't look like path-to-file (dump-heap app-name path-to-file false))) includes the app-name? Sorry for not seeing it, but where is that used in the file name? (obv you have generated these files so know what the name is)

Reply:
I reworked some of this (the logging) to be more clear, now all apps will report in the same way, not just the two I originally added the SIGTERM response to. The file name should be unique to application if your running multiple applications.

Reply:
Ok, thanks. But I think we will revisit this in the next ticket. Also, picking a random number between 0,1000 doesn't guarantee uniqueness so it may or may not actually generate 1000 discrete filenames. Timestamp may be the better way to go. 
>even if you're running multiple Tasks in AWS 
Note, it's very uncommon for us to only have a single task instance for any deployed CMR service in any environment. Bootstrap is the only one I can think of bc that's kinda treated as a singleton.